src/main/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/OrganisationService.java

```diff
@@ -0,0 +1,106 @@
+                .map(ProfileRefreshQueueEntity::getOrganisationProfileId)
+                .collect(Collectors.toList());
+        // HLD: Note that it is easier to take the maximum version number from step 2 and apply it to all organisations
+        // This is consistent with the semantics of "this version number or higher", and will cause no problems
+        Optional<Integer> maxVersion = profileRefreshQueueEntities.stream()
+                .map(ProfileRefreshQueueEntity::getAccessTypesMinVersion)
+                .max(Comparator.naturalOrder());
+
+        if (activeOrganisationProfileIds.isEmpty()) {
+            return;
+        }
+
+        OrganisationByProfileIdsRequest request = new OrganisationByProfileIdsRequest(activeOrganisationProfileIds);
+
+        OrganisationByProfileIdsResponse response;
+        response = prdService.fetchOrganisationsByProfileIds(Integer.valueOf(pageSize), null, request).getBody();
+
+        boolean moreAvailable;
+        String lastRecordInPage;
+
+        if (responseNotNull(response) && maxVersion.isPresent()) {
+            moreAvailable = response.getMoreAvailable();
+            lastRecordInPage = response.getLastRecordInPage();
+
+            insertIntoOrganisationRefreshQueue(response.getOrganisationInfo(), maxVersion.get());
+
+            while (moreAvailable) {
+                response = prdService.fetchOrganisationsByProfileIds(
+                        Integer.valueOf(pageSize), lastRecordInPage, request).getBody();
+
+                if (responseNotNull(response)) {
+                    moreAvailable = response.getMoreAvailable();
+                    lastRecordInPage = response.getLastRecordInPage();
+
+                    insertIntoOrganisationRefreshQueue(response.getOrganisationInfo(), maxVersion.get());
+                } else {
+                    break;
+                }
+            }
+
+            updateOrganisationRefreshQueueActiveStatus(activeOrganisationProfileIds, maxVersion.get());
+        }
+    }
+
+    private void insertIntoOrganisationRefreshQueue(List<OrganisationInfo> organisationInfo,
+                                                    Integer accessTypeMinVersion) {
+        organisationInfo.forEach(orgInfo -> organisationRefreshQueueRepository.insertIntoOrganisationRefreshQueue(
+                orgInfo.getOrganisationIdentifier(),
+                orgInfo.getLastUpdated(),
+                accessTypeMinVersion
+        ));
+    }
+
+    private void updateOrganisationRefreshQueueActiveStatus(List<String> organisationProfileIds,
+                                                            Integer accessTypeMaxVersion) {
+        organisationProfileIds.forEach(organisationProfileId -> profileRefreshQueueRepository.setActiveFalse(
+                organisationProfileId,
+                accessTypeMaxVersion
+        ));
+    }
+
+    private boolean responseNotNull(OrganisationByProfileIdsResponse response) {
```

Comment:
Last I looked there were some missing code coverage on the usage of this method

Reply:
<img width="413" alt="Screenshot 2024-02-09 at 14 55 22" src="https://github.com/hmcts/am-org-role-mapping-service/assets/130474274/121714e2-3ef5-4039-9afc-0d531f058040">

code code now at 100% for service.

i have removed `&& maxVersion.isPresent()` - as the check is redundant, as it's always `true` by the time the code hits the `if` statement. 

also modified `responseNotNull` method to just be `return response != null && !response.getOrganisationInfo().isEmpty();` these checks are sufficient so no need for additional `!response.getLastRecordInPage().isEmpty()
                && response.getMoreAvailable() != null` check